prometheus_hardware_exporter/collector.py

```diff
@@ -5,13 +5,134 @@
 
 from prometheus_client.metrics_core import GaugeMetricFamily, InfoMetricFamily
 
+from .collectors.perccli import PercCLI
 from .collectors.sasircu import LSISASCollectorHelper, Sasircu
 from .collectors.storcli import StorCLI
 from .core import BlockingCollector, Payload, Specification
 
 logger = getLogger(__name__)
 
 
+class PowerEdgeRAIDCollector(BlockingCollector):
+    """Collector for PowerEdge RAID controller."""
+
+    perccli = PercCLI()
+
+    @property
+    def specifications(self) -> List[Specification]:
+        return [
+            Specification(
+                name="perccli_command_success",
+                documentation="Indicates if the command is successful or not",
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="perccli_command_ctrl_success",
+                documentation="Indicates if the command for each controller is successful or not",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="poweredgeraid_controller_exists",
+                documentation="Indicates if the controller exists or not",
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="poweredgeraid_controllers",
+                documentation="Total number of controllers",
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="poweredgeraid_virtual_drive",
+                documentation="Number of virtual drives",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="poweredgeraid_virtual_drive_info",
+                documentation="Indicates the state of virtual drive",
+                labels=[
+                    "controller_id",
+                    "device_group",
+                    "virtual_drive_id",
+                    "state",
+                    "cache_policy",
+                ],
+                metric_class=GaugeMetricFamily,
```

Comment:
1. What's the different if we use InfoFamily? 
2. The name on the spec is "poweredgeraid_virtual_drive_state"

Reply:
1. As I understand from [here](https://github.com/prometheus/client_python/blob/master/prometheus_client/metrics_core.py#L359), it always return 1 but it will put the value as part of the labels. I will saw the result is the same because we already put all the items in the label and the value is already 1.
2. I haven't update spec after the change

Reply:
I prefer to align the practice in the team if possible. And I vote InfoMetricFamily, it makes the behavior (value is always 1) implicit.

Reply:
I update with InfoMetricFamily.